gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/components/SelectableAvatar.kt

```diff
@@ -4,82 +4,157 @@ import androidx.compose.foundation.background
 import androidx.compose.foundation.border
 import androidx.compose.foundation.clickable
 import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.BoxScope
 import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
 import androidx.compose.foundation.shape.RoundedCornerShape
 import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.IconButtonDefaults
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.draw.clip
 import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import coil.compose.AsyncImage
 import com.gravatar.quickeditor.R
 
+private val cornerRadius = 8.dp
+
 @Composable
 internal fun SelectableAvatar(
     imageUrl: String,
     isSelected: Boolean,
-    isLoading: Boolean,
+    loadingState: AvatarLoadingState,
     onAvatarClicked: () -> Unit,
     modifier: Modifier = Modifier,
+    onReloadClicked: () -> Unit = { },
 ) {
-    val cornerRadius = 8.dp
-    Box(modifier = modifier) {
+    Box(
+        modifier = modifier
+            .then(
+                if (isSelected) {
+                    Modifier.border(4.dp, MaterialTheme.colorScheme.primary, RoundedCornerShape(cornerRadius))
+                } else {
+                    Modifier.border(
+                        1.dp,
+                        MaterialTheme.colorScheme.surfaceDim,
+                        RoundedCornerShape(cornerRadius),
+                    )
+                },
+            )
+    ) {
```

Comment:
I extracted this border to the outer composable so that it's not covered by the overlay.

Reply:
This one caused all the updated to the screenshot tests. To be honest not sure why, but there's like a one pixel difference maybe - very hard to spot.